integrations/whatsapp/src/index.ts

```diff
@@ -12,15 +12,46 @@ import * as card from './message-types/card'
 import * as carousel from './message-types/carousel'
 import * as choice from './message-types/choice'
 import * as dropdown from './message-types/dropdown'
+import { checkManualConfiguration } from './misc/check-manual-config'
 import { getInterstitialUrl, redirectTo } from './misc/html-utils'
 import { getAccessToken, getPhoneNumberId, getSecret } from './misc/whatsapp'
 import { handleWizard } from './misc/wizard'
 import * as outgoing from './outgoing-message'
+import { identifyBot, trackIntegrationEvent } from './tracking'
 import { WhatsAppPayload } from './whatsapp-types'
 import * as bp from '.botpress'
 
 const integration = new bp.Integration({
-  register: async () => {},
+  register: async (input) => {
+    const { useManualConfiguration, accessToken, clientSecret, phoneNumberId, verifyToken } = input.ctx.configuration
+
+    await identifyBot(input.ctx.botId, {
+      [INTEGRATION_NAME + 'OauthType']: useManualConfiguration ? 'manual' : 'oauth',
+    })
+
+    if (!useManualConfiguration) {
+      return // nothing more to do if we're not using manual configuration
+    }
+
+    if (accessToken && clientSecret && phoneNumberId && verifyToken) {
+      // let's check the credentials
+      const manuallyConfigured = await checkManualConfiguration(accessToken)
+      if (!manuallyConfigured) {
+        await trackIntegrationEvent(input.ctx.botId, 'manualSetupStep', {
```

Comment:
Ah yes good point, the timeout option sounds good! I think we could even set it 2 or 3 seconds, should be fine as usually calling Segment should be pretty fast.

It has to do with the fact that AWS lambda by default doesn't check if there are pending events in the node event loop queue, it just waits for the lambda handler function to return.

Reply:
I put a 2 second max timeout and tested it. It seems good. Let me know if you want me to reduce it, or if you want to make changes on the infra-side to accomodate these.

Reply:
Thanks, looks good! Let me know once you're done with the changes so I can rereview the PR 🙂

Reply:
https://github.com/botpress/skynet/blob/387b29b57e7a63080cd3b21abcd25a92895877cf/packages/cdm-backend/src/sandbox/lambda_index.ts#L87-L88 I think this is what's causing me pain

Reply:
Correct, we do that by default for safety, for example to avoid rogue bots that could increase our lambda costs by spinning off async tasks outside of handling messages.